lib/go-tc/enum.go

```diff
@@ -105,6 +104,9 @@ func (t CacheType) String() string {
 }
 
 // CacheTypeFromString returns a cache type object from its string representation, or CacheTypeInvalid if the string is not a valid type.
+// Enums should always have a FromString() conversion function, to convert input
```

Comment:
The reason I distributed that comment to each applicable function is that, when it's at the top of the file, it ends up in the `tc` package overview. Compare the overview sections of https://godoc.org/github.com/apache/trafficcontrol/lib/go-tc and https://godoc.org/github.com/zrhoffman/trafficcontrol/lib/go-tc.

If we're fine with the overview section talking about (what is currently) a specific file, I can change it back.

Reply:
Well I'm not terribly opposed it's just that people likely won't copy the admonition when creating new enum types, and for what it's worth we already have the overview section talking about a specific file:

> _" enum.go contains enumerations and strongly typed names._
>
> _"These enums should be treated as enumerables, and MUST NOT be..."_

Reply:
That is true. Because those other lines in the overview mentioning enums were so long, it didn't make sense to copy those comments several times around the file. We might as well keep all of those notes in one place.

Addressed in 6c538637c8.